### PR TITLE
Reset Password API changes

### PR DESCRIPTION
**Pull request by BhargavaNarendra97**
## Release Notes

- **New Features**
	- Introduced self-service password reset functionality for users.
	- Added admin-specific password reset capability.
	- New endpoints for fetching school and user details.

- **Improvements**
	- Enhanced validation for password reset requests, accommodating different user roles.
	- Updated configuration properties for dynamic service URLs.
	- Updated password reset endpoint for improved specificity.

- **Bug Fixes**
	- Corrected password reset endpoint URLs in configuration files.

- **Documentation**
	- Updated internal documentation to reflect new functionalities and configurations.

### CODE REVIEW

File: src/main/java/com/mpsp/cc_auth_service/dto/SchoolDetails.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SchoolDetails {
+  private int schoolId;
+  private String schoolUdiseCode;
+  private Integer principalUserId;
```
[the remark this answers is not in the document]
Comment (BhargavaNarendra97) [reply src/main/java/com/mpsp/cc_auth_service/dto/SchoolDetails.java:14]: This is not required as it is not used by any request. This is a dto for response

File: src/main/java/com/mpsp/cc_auth_service/dto/ResetPasswordRequest.java [hunk 2]
```diff
@@ -10,13 +15,27 @@
...
 
+  @NotBlank(message = "User Role is required", groups = ResetPasswordByAdmin.class)
+  @Pattern(regexp = "STUDENT|TEACHER", message = "Invalid User Role")
+  private String behalfOfUserRole;
+
+  @NotNull(
+      message = "User Id of student or teacher is required",
+      groups = ResetPasswordByAdmin.class)
+  @Min(value = 1, message = "Invalid User Id", groups = ResetPasswordByAdmin.class)
+  private Integer behalfOf;
```
[the remark this answers is not in the document]
Comment (BhargavaNarendra97) [reply src/main/java/com/mpsp/cc_auth_service/dto/ResetPasswordRequest.java:37]: Max value is not requried. I prefer using STUDENT|TEACHER instead of enum

**Review by abhishekTraitFit — APPROVED** [review]

File: src/main/java/com/mpsp/cc_auth_service/config/SecurityConfig.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public SecurityFilterChain securityFilterChain(final HttpSecurity http) throws E
                         "/swagger-ui/**",
                         "/swagger-config/**",
                         "/api/v1/auth/forgot-password",
-                        "/api/v1/auth/reset-password",
+                        "/api/v1/auth/reset-password/self",
```
Comment (nnagwani-traitfit) [inline src/main/java/com/mpsp/cc_auth_service/config/SecurityConfig.java:42]: @BhargavaNarendra97 why are we removing the API to reset password for someone else? The admin / POC can still do it for Students? Teachers probably have a way to forget the password, students don't.

**Review by nnagwani-traitfit — APPROVED** [review]

File: src/main/java/com/mpsp/cc_auth_service/config/SecurityConfig.java (hunk 3 above)
Comment (nnagwani-traitfit) [reply src/main/java/com/mpsp/cc_auth_service/config/SecurityConfig.java:42]: nvm I see that you have both APIs

**Pull request merged**